app/controllers/test_reports_controller.rb

```diff
@@ -2,16 +2,21 @@
 
 class TestReportsController < ApplicationController
   before_action :check_round, only: %i[show update]
+  PER_PAGE = 30
   def index
-    per_page = 30
-    @page = Kaminari.paginate_array(Job.all_root_jobs.to_a, total_count: Job.all_root_jobs.length).page(params[:page]).per(per_page)
-    start_num = params[:page].nil? || params[:page] == 1 ? 0 : per_page * (params[:page].to_i - 1)
-    root_jobs = Job.root_jobs(start_num, per_page)
+    start_num = params[:page].nil? || params[:page] == 1 ? 0 : PER_PAGE * (params[:page].to_i - 1)
+    root_jobs, @page = if params[:search_word]
```

Comment:
If there is no `search_word` parameter,  `params[:search_word]` is nil?
If so, how about implementing it like this:

```ruby
          root_jobs if params[:search_word]
                      Job.get_searched_root_jobs(start_num, PER_PAGE, params[:search_word])
                    else
                      Job.root_jobs(start_num, PER_PAGE),
                    end

          @page = ganerate_pagenation(params[:search_word])
```

I also think you can remove the default parameters of `ganerate_pagenation` as well.